### PR TITLE
Update dependencies to use Go 1.21 & support VinylDNS 0.20.0

### PR DESCRIPTION
**Pull request by brucedewald**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions.

### Description of the Change

I plan to submit a PR to terraform-provider-vinyldns in the near future to add some features, but before adding any functionality, I wanted to get this repo & terraform-provider-vinyldns in good shape. This PR should not be changing any functionality, just updating several things as described here:
- Updating to use go 1.21 from 1.17 which was EOL'd in Aug 2022
- Replacing deprecated go-aws-auth package with the official AWS SDK for AWS SigV4 signing of requests
- Replacing deprecated ioutil package with io/os as appropriate
- Updated Makefile to use latest version of VinylDNS for integration testing
- Following changes to support version 0.20.0 of VinylDNS
  - Updated Makefile to properly start/stop test containers
  - Updating RecordSetChanges calls to set startfrom & nextID to be integers instead of strings
  - Minor changes to tests to reflect current state of VinylDNS

Note: I'm not sure if I should be bumping the version of this package as part of this PR?

I will also be submitting a similar PR to terraform-provider-vinyldns in the very near future with similar updates.

### Why Should This Be In The Package?

To ensure this package doesn't get further out of date.

### Benefits

Support of current VinylDNS and using stable version of Go.

### Possible Drawbacks

None to my knowledge.

### Verification Process

Ran `make all` which does go test and then spins up docker container with VinylDNS 0.20.0 and does integration testing.

```
~ make all
test -z ""
go vet ./...
GO111MODULE=on go test ./... -cover
ok      github.com/vinyldns/go-vinyldns/vinyldns        0.375s  coverage: 84.6% of statements
GO111MODULE=on go build -ldflags "-X main.version=0.9.16" ./...
/Users/bdewald/go/1.21.9/src/github.com/vinyldns/vinyldns-0.20.0/quickstart/quickstart-vinyldns.sh \
                --clean
Killing running containers...
76382067785c
ea00febbe124
Removing containers...
76382067785c
ea00febbe124
Deleted Networks:
vinyldns_net

Clean up completed!
if [ ! -d "/Users/bdewald/go/1.21.9/src/github.com/vinyldns/vinyldns-0.20.0" ]; then \
                echo "github.com/vinyldns/vinyldns-0.20.0 not found in your GOPATH (necessary for acceptance tests), getting..."; \
                git clone \
                        --branch v0.20.0 \
                        https://github.com/vinyldns/vinyldns \
                        /Users/bdewald/go/1.21.9/src/github.com/vinyldns/vinyldns-0.20.0; \
        fi
/Users/bdewald/go/1.21.9/src/github.com/vinyldns/vinyldns-0.20.0/quickstart/quickstart-vinyldns.sh \
                --api \
                --version-tag 0.20.0
Starting VinylDNS (latest) the background...
[+] Running 2/0
[+] Running 2/5ldns_net                                                                                                                                       Created0.1s 
[+] Running 3/5ldns_net                                                                                                                                       Created0.2s 
[+] Running 4/5ldns_net                                                                                                                                       Created0.3s 
 ⠸ Network vinyldns_net                                                                                                                                       Created0.4s 
 ✔ Container vinyldns-api-integration                                                                                                                         Started0.2s 
 ! integration The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested        0.0s 
 ✔ Container vinyldns-api                                                                                                                                     Started0.3s 
 ! api The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested                0.0s 

Waiting for VinylDNS API at http://localhost:9000 ................OK

VinylDNS API started! You can connect to the API via http://localhost:9000
To clean up the running containers:
    /Users/bdewald/go/1.21.9/src/github.com/vinyldns/vinyldns-0.20.0/quickstart/quickstart-vinyldns.sh --clean
GO111MODULE=on go test ./... -tags=integration
ok      github.com/vinyldns/go-vinyldns/vinyldns        43.921s
cat vinyldns/version.go | grep 'var Version = "0.9.16"'
var Version = "0.9.16"
GO111MODULE=on go install ./...
```

### Applicable Issues (Optional)

N/A

### CODE REVIEW

File: Makefile
Comment (Jay07GIT) [inline Makefile:4]: We are trying to update Vinyldns api version to latest in all the clients. If the same changes added here, it would be appreciated.

**Review by Jay07GIT — CHANGES_REQUESTED** [review]

File: Makefile
Comment (brucedewald) [reply Makefile:4]: Hi @Jay07GIT  I did set this to 0.20.0 already which is the latest version of Vinyldns here https://github.com/vinyldns/vinyldns/releases

or am I misunderstanding your ask?

Comment (brucedewald) [reply Makefile]: @Jay07GIT  - just pushed a commit where I changed it to ensure it's always pulling the current from master branch of vinyldns before doing integration tests. I don't want to make changes here to switch to using docker image directly as that would require including dockerfile & bind configs in this repo then and I'm trying to keep this specific PR to just updating things with same functionality. Happy to improve in future PRs though.

File: vinyldns/endpoints.go
Comment (Aravindh-Raju) [inline vinyldns/endpoints.go:70]: We can entirely remove the `nameFilter` parameter as there isn't such filter for recordsetChanges as of now. We can just have:
```
query := buildQueryInt(f)
```

File: vinyldns/endpoints.go
Comment (Aravindh-Raju) [inline vinyldns/endpoints.go:142]: Same as before. We don't have a `nameFilter` param for recordsetChange. We do have two new parameters that were added (`fqdn` and `recordType`) which is used while getting a record's change history: https://github.com/vinyldns/vinyldns/blob/master/modules/docs/src/main/mdoc/api/get-recordset-change-history.md. 

We still have some changes to make there, so we can remove the `nameFilterName` parameter and this if block for now.

File: vinyldns/recordsets_resources.go
Comment (Aravindh-Raju) [inline vinyldns/recordsets_resources.go:31]: Since startFrom is `int` type, we can replace the `any` with `int`.

File: vinyldns/resources.go
Comment (Aravindh-Raju) [inline vinyldns/resources.go:57]: I don’t think `ListFilterInt` is used anywhere else other than for recordsetChanges, and it doesn't have any nameFilter as of now. We can remove this for now.

**Review by Aravindh-Raju — CHANGES_REQUESTED** [review]
Some minor changes. 

We will later have to change the `buildQueryInt` function to something like [buildGlobalListQuery](https://github.com/vinyldns/go-vinyldns/blob/main/vinyldns/endpoints.go#L136) to include `fqdn` and `recordType` parameters that we now have for record's change history

**Comment by Aravindh-Raju** [comment]
@brucedewald, we will wait for other's review and decide if the above changes were necessary.

File: Makefile
Comment (brucedewald) [reply Makefile]: @Jay07GIT  - just updated it so it defaults to latest but you can easily change the variable in the makefile if you want to test a specific version for testing

File: vinyldns/endpoints.go
Comment (brucedewald) [reply vinyldns/endpoints.go]: @Aravindh-Raju - Thanks for the feedback! I actually went ahead and removed namefilter as suggested and also renamed `ListFilterInt` -> `ListFilterRecordSetChanges` and `buildQueryInt` -> `buildRecordSetChangesQuery` to be clearer as to what they're used for.

File: Makefile
Comment (Jay07GIT) [inline Makefile:30]: Everything looks good to me expect this minor typo change.

Kindly update clonevinyl to clone-vinyl.

**Review by Jay07GIT — CHANGES_REQUESTED** [review]

File: Makefile
Comment (brucedewald) [reply Makefile]: @Jay07GIT corrected

**Review by Jay07GIT — APPROVED** [review]
LGTM.

**Comment by brucedewald** [comment]
Thank you @Aravindh-Raju @Jay07GIT  for your reviews & approvals - what is needed to get this PR merged and a new release tag published?

**Comment by brucedewald** [comment]
@Aravindh-Raju @Jay07GIT - I wanted to follow up on my previous comment to see if you could provide any guidance here, thanks! Another PR is pending on a new release being created from this one: https://github.com/vinyldns/terraform-provider-vinyldns/pull/116

Additionally, I plan to submit 1 or more feature PRs once this initial "clean up" work is done. Thank you!

**Review by nspadaccino — APPROVED** [review]
Looks good

**Review by arpit4ever — APPROVED** [review]
LGTM

**Review by nspadaccino — CHANGES_REQUESTED** [review]
Hey @brucedewald, this repo requires all commits to be signed via GPG key, I see that your latest commit is signed, but the commits before that aren't, which is blocking us from merging. You can retroactively sign your unsigned commits using this command:

GIT_EDITOR=true git rebase --exec 'git commit --amend --no-edit -n -S' -i LAST_COMMIT_HASH_YOU_WANT_TO_SIGN

You can locally check if your commits are signed or not using
git log --show-signature

After you've rebased, you'll have to force-push the branch, since the commit history will have been edited.

**Comment by brucedewald** [comment]
Thanks @nspadaccino - All commits show as verified now

**Review by nspadaccino — APPROVED** [review]

**Pull request merged**
